test/assets.js

```diff
@@ -284,10 +293,222 @@ describe('Assets', function() {
     assert(bundlePage.includes(getScriptMarkup('extra2-module-bundle', 'module')));
     assert(bundlePage.includes(getScriptMarkup('extra2-nomodule-bundle', 'nomodule')));
   });
+
+  it('should build with cache and gain performance', async function() {
+    await t.destroy(apos);
+    await removeCache();
+    await removeCache(cacheFolderPath.replace('/webpack-cache', '/changed'));
+
+    const es5Modules = {
+      ...modules,
+      '@apostrophecms/asset': {
+        options: {
+          es5: true
+        }
+      }
+    };
+
+    apos = await t.create({
+      root: module,
+      modules: es5Modules
+    });
+    assert.throws(() => fs.readdirSync(cacheFolderPath), {
+      code: 'ENOENT'
+    });
+
+    let startTime;
+
+    // Cold run
+    startTime = Date.now();
+    await apos.asset.tasks.build.task({
+      'check-apos-build': false
+    });
+    const execTime = Date.now() - startTime;
+    const { meta, folders } = getCacheMeta();
+    assert.equal(folders.length, 3);
+    assert.equal(Object.keys(meta).length, 3);
+    assert(meta['default:apos']);
+    assert(meta['default:src']);
+    assert(meta['default:src-es5']);
+
+    // Cache
+    startTime = Date.now();
+    await apos.asset.tasks.build.task({
+      'check-apos-build': false
+    });
+    const execTimeCached = Date.now() - startTime;
+    const { meta: meta2, folders: folders2 } = getCacheMeta();
+    assert.equal(folders2.length, 3);
+    assert.equal(Object.keys(meta2).length, 3);
+    assert(meta2['default:apos']);
+    assert(meta2['default:src']);
+    assert(meta2['default:src-es5']);
+
+    // Expect at least 40% gain, in reallity it should be 50+
+    const gain = (execTime - execTimeCached) / execTime * 100;
+    assert(gain >= 40, `Expected gain >=40%, got ${gain}%`);
```

Comment:
This test is really cool but I don't think we can have such a strict standard for something that is highly dependent on the performance given to us by github actions from moment to moment. We will have too many false positives / negatives. This test could be enabled by an environment variable for local use.

Reply:
Maybe set the expectation to 20% and we'll see if that is stable enough.